src/test/java/nextstep/sessions/infrastructure/SessionRepositoryTest.java

```diff
@@ -0,0 +1,135 @@
+    }
+
+
+    @Test
+    void 강의_생성() {
+        EnrollmentInfo enrollmentInfo = new EnrollmentInfo(new SessionType(PaidType.PAID, 800000L, 10), SessionState.RECRUITING);
+        OpenInfo openInfo = new OpenInfo(new Duration(LocalDateTime.now(), LocalDateTime.now().plusDays(10)));
+        SessionInfo sessionInfo = new SessionInfo(enrollmentInfo, openInfo);
+        Session session = new Session(sessionInfo);
+
+        int count = sessionRepository.saveSession(session);
+        assertThat(count).isEqualTo(1);
+
+        Session savedSession = sessionRepository.findSessionBySessionId(1);
+        assertThat(savedSession.sessionInfo().enrollmentInfo().sessionState()).isEqualTo(session.sessionInfo().enrollmentInfo().sessionState());
+    }
+
+    @Test
+    void 커버_이미지_생성() {
+        CoverImage coverImage = new CoverImage("image.png", 102400, 300, 200);
+
+        int count = sessionRepository.saveCoverImage(1, coverImage);
+        assertThat(count).isEqualTo(1);
+
+        CoverImage savedCoverImage = sessionRepository.findCoverImageBySessionId(1);
+        assertThat(savedCoverImage.imageType()).isEqualTo(coverImage.imageType());
+    }
+
+    @Test
+    void 유료_강의_수강_신청_성공() {
+        int sessionId = 6;
+        Session session = sessionRepository.findSessionBySessionId(sessionId);
+        List<Registration> registrations = sessionRepository.findAllRegistrations(sessionId);
+        Session sessionWithRegistrations = new Session(session.sessionInfo(), registrations);
+
+        sessionRepository.saveRegistration(sessionId, sessionWithRegistrations.registration(NsUserTest.NEWJIGI, new Payment(1L, 2L, 3L, 800000L)));
+        List<Registration> current_registrations = sessionRepository.findAllRegistrations(sessionId);
+        assertThat(current_registrations).hasSize(registrations.size() + 1);
+    }
+
+    @Test
+    void 무료_강의_수강_신청_성공() {
+        int sessionId = 7;
+        Session session = sessionRepository.findSessionBySessionId(sessionId);
+        List<Registration> registrations = sessionRepository.findAllRegistrations(sessionId);
+        Session sessionWithRegistrations = new Session(session.sessionInfo(), registrations);
+
+        sessionRepository.saveRegistration(sessionId, sessionWithRegistrations.registration(NsUserTest.NEWJIGI, new Payment(1L, 2L, 3L, 800000L)));
+        List<Registration> current_registrations = sessionRepository.findAllRegistrations(sessionId);
+        assertThat(current_registrations).hasSize(registrations.size() + 1);
+    }
+
+    @Test
+    void 수강_신청_실패_인원_초과() {
+        int sessionId = 10;
+        Session sessionWithRegistrations = getSession(sessionId);
+
+        assertThatThrownBy(() -> sessionRepository.saveRegistration(sessionId, sessionWithRegistrations.registration(NsUserTest.NEWJIGI, new Payment(1L, 2L, 3L, 800000L))))
+            .isInstanceOf(SessionsException.class)
+            .hasMessage("강의 최대 인원을 초과했습니다.");
+    }
```

Comment:
repository 테스트와 목적이 조금 다른 것 같은데요
SessionRepository#saveRegistration 메서드를 테스트할 것이라 생각했는데
Session#registration 메서드에 대한 테스트인 것 같아요

saveRegistration 메서드를 테스트하기 위해 아래처럼 registration 결과를 지역 변수로 추출하면 테스트는 실패해요 😢 
```suggestion
    @Test
    void 수강_신청_실패_인원_초과() {
        int sessionId = 10;
        Session sessionWithRegistrations = getSession(sessionId);
        final Registration registration = sessionWithRegistrations.registration(NsUserTest.NEWJIGI, new Payment(1L, 2L, 3L, 800000L));

        assertThatThrownBy(() -> sessionRepository.saveRegistration(sessionId, registration))
            .isInstanceOf(SessionsException.class)
            .hasMessage("강의 최대 인원을 초과했습니다.");
    }
```

Reply:
그렇네요..
수강 신청 메서드 실행 후 테스트 항목을 추가했습니다.
- 수강 신청 전 목록 개수 + 1 확인
- 수강 신청한 유저가 신청 후 목록에 포함되어 있는 지 확인